From 0d0852b553fbc7f7627648373cd3f064c5c4d41c Mon Sep 17 00:00:00 2001
From: Joseph Weitekamp <jweite@amazon.com>
Date: Thu, 10 Sep 2020 23:56:07 -0400
Subject: [PATCH] Change to test_s3 method
 test_presigned_url_restrict_parameters to tolerate change in exception
 messages, spurred by boto3 1.14.59 release.

---
 tests/test_s3/test_s3.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/test_s3/test_s3.py b/tests/test_s3/test_s3.py
index 960594801..1a8b766e9 100644
--- a/tests/test_s3/test_s3.py
+++ b/tests/test_s3/test_s3.py
@@ -4655,8 +4655,8 @@ def test_presigned_url_restrict_parameters():
             ClientMethod="put_object",
             Params={"Bucket": bucket, "Key": key, "Unknown": "metadata"},
         )
-    assert str(err.exception).should.equal(
-        'Parameter validation failed:\nUnknown parameter in input: "Unknown", must be one of: ACL, Body, Bucket, CacheControl, ContentDisposition, ContentEncoding, ContentLanguage, ContentLength, ContentMD5, ContentType, Expires, GrantFullControl, GrantRead, GrantReadACP, GrantWriteACP, Key, Metadata, ServerSideEncryption, StorageClass, WebsiteRedirectLocation, SSECustomerAlgorithm, SSECustomerKey, SSECustomerKeyMD5, SSEKMSKeyId, SSEKMSEncryptionContext, RequestPayer, Tagging, ObjectLockMode, ObjectLockRetainUntilDate, ObjectLockLegalHoldStatus'
+    assert str(err.exception).should.match(
+        r'Parameter validation failed:\nUnknown parameter in input: "Unknown", must be one of:.*'
     )
 
     s3.delete_bucket(Bucket=bucket)
